test/parallel/test_tensorflow.py

```diff
@@ -4085,8 +4083,11 @@ def test_horovod_syncbn_gpu(self):
             for x in x_list:
                 bn = tf.keras.layers.BatchNormalization(axis=1, fused=False)
                 sync_bn = hvd.SyncBatchNormalization(axis=1)
-                bn_func = bn.apply(x, training=True)
-                sync_bn_func = sync_bn.apply(tf.expand_dims(x[hvd.rank()], 0), training=True)
+                # apply() API is removed in TF 2.9.0
+                bn_func = bn.apply(x, training=True) if LooseVersion(
+                    tf.__version__) < LooseVersion('2.9.0') else bn(x, training=True)
+                sync_bn_func = sync_bn.apply(tf.expand_dims(x[hvd.rank()], 0), training=True) if LooseVersion(
+                    tf.__version__) < LooseVersion('2.9.0') else sync_bn(tf.expand_dims(x[hvd.rank()], 0), training=True)
```

Comment:
The code could be simplified as follows:

```suggestion
                if LooseVersion(tf.__version__) < LooseVersion('2.9.0'):
                    bn = bn.apply
                    sync_bn = sync_bn.apply
                bn_func = bn(x, training=True)
                sync_bn_func = sync_bn(tf.expand_dims(x[hvd.rank()], 0), training=True)
```